packages/core/core/src/services/document-service/repository.ts

```diff
@@ -193,7 +194,24 @@ export const createContentTypeRepository: RepositoryFactoryMethod = (
         // assign new documentId
         assoc('documentId', createDocumentId()),
         // Merge new data into it
-        (data) => merge(data, queryParams.data),
+        async (data) =>
+          /**
+           * Merge receives [{ connect: [{ id: 1 }] }] and { connect: [{ documentId }] }]
+           * but it does not merge arrays, so only keeps one of them.
+           * We want to not do connect for everything, but use both connect (for the new one) and
+           * set (for the old one) at the same time. This requires a change at the database level.
+           * Can this be done?
+           */
```

Comment:
If I recall correctly, the limitation here was that using "set" for the old could stole relations, and if the user disconnected some relations to prevent that, they would still be cloned.